SU2_CFD/src/output/COutput.cpp

```diff
@@ -1504,66 +1524,85 @@ void COutput::PreprocessVolumeOutput(CConfig *config){
 
   SetVolumeOutputFields(config);
 
-  /*---Coordinates and solution groups must be always in the output.
-   * If they are not requested, add them here. ---*/
-
+  /*--- Coordinates must be always in the output.
+   * If they are not requested, add them here.
+   * SOLUTION is only required for restart files. ---*/
   auto itCoord = std::find(requestedVolumeFields.begin(),
                                           requestedVolumeFields.end(), "COORDINATES");
-  if (itCoord == requestedVolumeFields.end()){
+  if (itCoord == requestedVolumeFields.end()) {
     requestedVolumeFields.emplace_back("COORDINATES");
     nRequestedVolumeFields++;
   }
-  auto itSol = std::find(requestedVolumeFields.begin(),
-                                          requestedVolumeFields.end(), "SOLUTION");
-  if (itSol == requestedVolumeFields.end()){
+
+  /*--- If no volume fields were requested, we add the entire SOLUTION field.
+   *    We also add the solution field if we are not using the compact formulation.
+   *    This is for backwards compatibility. ---*/
+  if ((config->GetWrt_Restart_Compact() == false) || (nRequestedVolumeFields == 1)) {
     requestedVolumeFields.emplace_back("SOLUTION");
     nRequestedVolumeFields++;
   }
```

Comment:
This logic is not correct, the idea was to add the solution if it was not present in the list of quantities, what you have is just if the list is empty which is generally not the case because derived classes add their own default.

I recommend keeping the old implementation because certain important quantities are part of the "solution", for example density.

Reply:
My intention was that the user can choose to write only a single field to e.g. a paraview file, previously the SOLUTION was always written to file.
The other derived classes actually already do a check on _(nRequestedVolumeFields == 0)_ , this one used to specifically check for the existence of the SOLUTION field. I think it is now more compatible with the other implementation as well, we can also remove the condition that we only do this if we ask for compact restarts. It is just my attempt to make it a bit more backward compatible.

Reply:
OK I see your point regarding SOLUTION vs PRIMITIVE, but that is why I had the extra condition that this behavior is only active when a user is using the compact_restart keyword, nothing changes for the user and he can only get this behavior when he actively sets the keyword COMPACT_RESTART.
But maybe we can think of a better way to get this behavior later in another PR? At least I would like to have the ability to save paraview files that only contain a single field.

Reply:
Yep but I liked this idea of compact restarts now that we don't rely on SU2_SOL for anything, so making it the default makes sense.
Maybe we can add some special keyword to the output field option, for example NO_DEFAULTS, and then you get exclusively what you ask for?

Reply:
OK, so remove the `((config->GetWrt_Restart_Compact() == false)` here and replace it with checking for RequestedVolumeField=="NO_DEFAULTS"?

Reply:
Yes, if that field is present, do not add the solution by default, otherwise keep what we had

Reply:
OK, I added the keyword COMPACT.